src/schematools/types.py

```diff
@@ -485,12 +495,13 @@ def get_fields_by_id(self, *field_ids: str) -> List[DatasetFieldSchema]:
         return [field for field in self.fields if field.id in field_ids_set]
 
     @lru_cache()  # type: ignore[misc]
-    def get_field_by_id(self, field_id: str) -> Optional[DatasetFieldSchema]:
+    def get_field_by_id(self, field_id: str) -> DatasetFieldSchema:
         """Get a fields based on the ids of the field."""
         for field_schema in self.fields:
             if field_schema.id == field_id:
                 return field_schema
-        return None
+
+        raise SchemaObjectNotFound(f"Field '{field_id}' does not exist in table '{self.id}'.")
```

Comment:
There's logic that depends on `get_field_by_id` returning `None` and not expecting any exceptions to be raised. 
e.g: https://github.com/Amsterdam/schema-tools/blob/92d86b1c1b08ea97819d8983b91146fd07fb29a1/src/schematools/contrib/django/factories.py#L86

Reply:
Fortunately that code did raise an exception:

```
        if sequence_field is None or identifier_field is None:

            raise ValueError(f"Cannot find temporal fields of table {related_table.id}")
 ```

so that is no longer needed, as the call already aborts.